resin/exceptions.py

```diff
@@ -310,6 +310,60 @@ class UnsupportedFunction(Exception):
     """
 
     def __init__(self, required_version, current_version):
-        self.code = 'UnsupportedFunction'
+        self.code = self.__class__.__name__
```

Comment:
Nit: Should this (both `self.code` and `self.exit_code`) live in some kind of `ResinException` superclass?